### ISSUE TITLE
How fasttext supervised train/updating weight of the model?

### ISSUE THREAD

**Issue by kurniawanen** [facebookresearch/fastText#798]
Could someone help me understand and clarify things on training and predict on fasttext supervised?

- Fasttext train exactly like word2vec (per window training);
I notice there is a ‘context window’ option in fasttext supervised. Also, from the paper I read that fasttext supervised architecture is like word2vec CBOW but the target word is changed into the label. So for each sentence, the model is trained for multiple times (correspond to the size of the context window)?

- Backpropagation also is done in word embedding weight, not just in the linear classifier;
My understanding is the word embedding and classifier is in one architecture. Obviously, we update the linear classifier weight using backpropagation. We also calculate the backpropagation to update the word vector right?   (weight of embedding layer).

- Implementation of predict is multiplication of word vector inputted in linear classifier;
For each word in the sentence, get the vector, sum it all, then divide by the number of words. After we got this averaged vector, multiply it by linear classifier weight for each corresponding neuron, calculate the softmax, and we got the prediction result. Is that it?

Did I miss something?
